modules/base/rendering/screenspaceinsetblackout.cpp

```diff
@@ -0,0 +1,768 @@
+        corners->dataHasChanged ||
+        topSpline->dataHasChanged ||
+        rightSpline->dataHasChanged ||
+        bottomSpline->dataHasChanged ||
+        leftSpline->dataHasChanged;
+}
+
+void ScreenSpaceInsetBlackout::BlackoutShape::resetHasChanged() {
+    textureTypeHasChanged = false;
+    corners->dataHasChanged = false;
+    topSpline->dataHasChanged = false;
+    rightSpline->dataHasChanged = false;
+    bottomSpline->dataHasChanged = false;
+    leftSpline->dataHasChanged = false;
+}
+
+void ScreenSpaceInsetBlackout::BlackoutShape::checkAndUpdateGUI() {
+    std::array<std::reference_wrapper<Spline*>,4> refs = {
+        std::ref(topSpline),
+        std::ref(rightSpline),
+        std::ref(bottomSpline),
+        std::ref(leftSpline)
+    };
+
+    // Check which splines that needs to be updated and perform correct actions
+    // Note: Need to remove propertySubOwner and add it again for GUI to work properly
+    bool updatePropertyTree = false;
+    for (Spline*& spline : refs) {
+        if (spline->pointAdded) {
+            std::string baseString = spline->base;
+            std::vector<glm::vec2>& dataRef = spline->data;
+            removePropertySubOwner(spline);
+            spline->addPoint();
+            delete spline;
+            spline = new Spline(dataRef, baseString);
+            spline->dataHasChanged = true;
+            updatePropertyTree = true;
+        }
+        else if (spline->pointRemoved) {
+            std::string baseString = spline->base;
+            std::vector<glm::vec2>& dataRef = spline->data;
+            removePropertySubOwner(spline);
+            spline->removePoint();
+            delete spline;
+            spline = new Spline(dataRef, baseString);
+            spline->dataHasChanged = true;
+            updatePropertyTree = true;
+        }
+    }
+
+    // Remove GUI elements so that we can add them in correct order again
+    if (updatePropertyTree) {
+        std::vector<openspace::properties::PropertyOwner*> subs = propertySubOwners();
+        for (int i = 0; i < subs.size(); i++) {
+            removePropertySubOwner(subs[i]);
+        }
+        addPropertySubOwner(corners);
+        for (Spline*& spline : refs) {
+            addPropertySubOwner(spline);
+        }
+    }
+}
+
+void ScreenSpaceInsetBlackout::BlackoutShape::copyToClipboard() {
+    std::string strCorners = formatLine("Corners", cornerData);
+    std::string strTop = formatLine("Top", topSplineData);
+    std::string strRight = formatLine("Right", rightSplineData);
+    std::string strBottom = formatLine("Bottom", bottomSplineData);
+    std::string strLeft = formatLine("Left", leftSplineData);
+
+    ghoul::setClipboardText(strCorners + strTop + strRight + strBottom + strLeft);
+}
+
+std::string ScreenSpaceInsetBlackout::BlackoutShape::formatLine(std::string id,
+                                                    const std::vector<glm::vec2>& data)
+{
+    std::string str = std::format("{} = {{ ", id);
```

Comment:
I think it would be great if this function would return `""` if `data` is empty to minimize the copy-paste text and not have anything in there that the user didn't set manually